### PR TITLE
Handle gzip encoded response bodies

### PR DESCRIPTION
**Pull request by Ripolin**
If PR #419 works for zlib encoded response with a "deflate" HTTP "Content-Encoding" header, I see anything about the "gzip" encoding ([www.gzip.org/zlib/zlib_faq.html#faq38](www.gzip.org/zlib/zlib_faq.html#faq38)).

My Apache HTTP Server 2.4 only deliver gzip content and when I use k6 (without this fix) to test some URLs, I've got binary data instead of JSON content.

### CODE REVIEW

File: js/modules/k6/http/http_request.go [hunk 1]
```diff
@@ -265,6 +266,9 @@ func (h *HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.Stat
 		if res.Header.Get("Content-Encoding") == "deflate" {
 			res.Body, resErr = zlib.NewReader(res.Body)
 		}
+		if res.Header.Get("Content-Encoding") == "gzip" {
```
Comment (robingustafsson) [inline js/modules/k6/http/http_request.go]: Please make this an `else if`.

Comment (Ripolin) [reply js/modules/k6/http/http_request.go]: Thanks for your reply. A "switch" statement could it be better ?

Comment (robingustafsson) [reply js/modules/k6/http/http_request.go]: Yup, a switch statement sounds good.

**Comment by Ripolin** [comment]
Done !

**Comment by robingustafsson** [comment]
Great @Ripolin!

Sorry, I should have been more explicit with the test change, I meant adding tests (in addition to) the current test cases that check that gzip/deflate works without the `Accept-Encoding` header explicitly set, since that is what I had misunderstood previously that Go net/http handled the two cases differently. So just want to make sure we test both cases, with and without that header set. If you fix that we're ready to merge!

**Comment by Ripolin** [comment]
Tests added !

**Comment by robingustafsson** [comment]
Thanks @Ripolin!

**Pull request merged**